src/dotnet/ToolPackage/IToolPackageInstaller.cs

```diff
@@ -9,12 +9,14 @@ namespace Microsoft.DotNet.ToolPackage
 {
     internal interface IToolPackageInstaller
     {
-        IToolPackage InstallPackage(
-            string packageId,
-            string packageVersion = null,
+        IToolPackage InstallPackage(NuGetPackageLocation nuGetPackageLocation,
             string targetFramework = null,
-            FilePath? nugetConfig = null,
-            string source = null,
             string verbosity = null);
+
+        IReadOnlyList<CommandSettings>
+            InstallPackageToNuGetCache(NuGetPackageLocation nuGetPackageLocation,
```

Comment:
It's a little weird to me that this method returns a collection of commands and the other one returns a "package", but I understand that in this case we're not really the ones maintaining the packages when we're using the nuget cache.